### ISSUE TITLE
R lgb.train crashing on simple example

### ISSUE THREAD

**Issue by zkurtz** [Microsoft/LightGBM#963]
```
rm(list = ls())
library(lightgbm)
library(data.table)

n = 1000
set.seed(0)

# Simulate some data
label = rbinom(n, size = 1, prob = 0.5)
features = data.table(
    v1 = sample(1:10, size = n, replace = TRUE),
    v2 = sample(1:10, size = n, replace = TRUE)
    )
features = as.matrix(lightgbm::lgb.prepare2(features))
train = lightgbm::lgb.Dataset(data = features,
                              label = labels)

# Choose parameters
params = list(
        objective = "binary",
        num_leaves = 20L,
        learning_rate = 0.1,
        min_data=1L
    )

# Boosting
bst = lightgbm::lgb.train(
    params = params, 
    data = train, 
    nrounds = 40L,
    verbose = 0)
```
The final line returns
```
Error in as.numeric(info) : 
  cannot coerce type 'closure' to vector of type 'double'
```
This behavior was the same on Mac Sierra and on Red Hat linux

**Comment by guolinke** [comment]
@zkurtz 
I am not familiar with R, but it seems the type of label is wrong. 
@Laurae2 @bfgray3

**Comment by bfgray3** [comment]
I believe the error is in the following line: https://github.com/Microsoft/LightGBM/blob/7193c30fb545306c9e8bb02e9a428fb3fc0a02c1/R-package/R/lgb.Dataset.R#L429.  This is unrelated to my recent pull request, but I'm wondering if these few lines should be rewritten as:

```
          ret <- if (name == "group") {
            integer(info_len) # Integer
          } else {
            numeric(info_len) # Numeric
          }
```

as appears a few lines previous.  Right now, R is treating `info` as a function (closure), and that's why it can't turn it into a numeric vector.

**Comment by bfgray3** [comment]
@zkurtz when you create the response variable you name it `label` but then in the `lgb.Dataset` call you use `labels`, which is a function from the `base` package. Please rerun, removing the "s" from the variable name and see if that resolves the issue.

**Comment by zkurtz** [comment]
@bfgray3 You are correct -- I have a basic typo in my example, and replacing `labels` with `label` fixes one issue. Now I don't what to call this (a feature request?) but it seems like the error message could be more helpful based on some kind of check of the input label object type.  I imagine there should be some list of acceptable types that would include neither `closure` nor `function`:

```
> class(labels)
[1] "function"
> typeof(labels)
[1] "closure"
```
